packages/core/src/tools/grep.ts

```diff
@@ -23,7 +25,114 @@ import { getErrorMessage, isNodeError } from '../utils/errors.js';
 import { isGitRepository } from '../utils/gitUtils.js';
 import { Config } from '../config/config.js';
 
-// --- Interfaces ---
+/**
+ * Memory safety constants to prevent heap overflow
+ */
+const MEMORY_SAFETY = {
```

Comment:
nit: instead of calling these MEMORY_SAFETY pick a more generic name that indicates these are the limits placed on the grep tool.